pkg/security/admission/admission.go

```diff
@@ -164,7 +165,25 @@ func assignSecurityContext(provider scc.SecurityContextConstraintsProvider, pod
 
 	errs := fielderrors.ValidationErrorList{}
 
+	psc, err := provider.CreatePodSecurityContext(pod)
+	if err != nil {
+		errs = append(errs, fielderrors.NewFieldInvalid("spec.securityContext", pod.Spec.SecurityContext, err.Error()))
+	}
+
+	// save the original PSC and validate the generated PSC.  Leave the generated PSC
+	// set for container generation/validation.  We will reset to original post container
+	// validation.
+	originalPSC := pod.Spec.SecurityContext
+	pod.Spec.SecurityContext = psc
+	errs = append(errs, provider.ValidatePodSecurityContext(pod).Prefix("spec.securityContext")...)
+
 	for i, c := range pod.Spec.Containers {
```

Comment:
Ok, looking through this loop, its really hard to read what's modifying a container and what is modifying a copy of a container that is going to be discarded later.

Re-reading, I **think** that this use a pointer to a copy of a container to perform a bunch of mutations that are ultimately discarded.  If that's right, can you rename `c` to `containerCopy` or some such?